redun/file.py

```diff
@@ -691,10 +721,46 @@ def shell_copy(
         recursive : bool
             If True, copy a directory tree of files.
         """
+
+        def to_mount_directory(path):
+            parsed_path = urlparse(path)
+            return f"/mnt/disks/share/{parsed_path.netloc}{parsed_path.path}"
+
+        if as_mount:
+            src_proto, dest_proto = get_proto(src_path), get_proto(dest_path)
+            if src_proto == "gs":
+                src_path = to_mount_directory(src_path)
+            if dest_proto == "gs":
+                dest_path = to_mount_directory(dest_path)
+
+            # When staging mounted files, soft-link to stage in.
+
+            if dest_path:
+                # Ensure dest path exists.
+                mk_dest_dir = f"mkdir -p {quote(os.path.dirname(dest_path))}"
+
+            if src_path and dest_path:
+                if src_proto == "gs" and dest_proto == "local":
+                    return f"ln -s {quote(src_path)} {quote(dest_path)}"
+                if src_proto == "local" and dest_proto == "gs":
+                    if not recursive:
+                        return f"{mk_dest_dir} && cp {quote(src_path)} {quote(dest_path)}"
+                    else:
+                        return f"{mk_dest_dir} && cp -r {quote(src_path)} {quote(dest_path)}"
+            elif recursive:
+                raise ValueError("recursive is not supported with stdin or stdout.")
+            elif src_path:
+                return f"cat {quote(src_path)}"
+            elif dest_path:
+                return f"$({mk_dest_dir}) cat - > {quote(dest_path)}"
```

Comment:
Perhaps we add a small comment so that future devs understand why we're avoiding `&&` like the other commands. Something like:

```suggestion
                # We use a subshell to make dest_dir so that cat receives stdin.
                return f"$({mk_dest_dir}) cat - > {quote(dest_path)}"
```

Reply:
Updated with this change.